tests/Offer/ReadModel/JSONLD/MediaUrlOfferRepositoryDecoratorTest.php

```diff
@@ -0,0 +1,133 @@
+                'contentUrl' => 'https://io.uitdatabank.be/images/21defff6-59e4-49f4-ab84-83302ff20010.jpeg',
+                'thumbnailUrl' => 'https://io.uitdatabank.be/images/21defff6-59e4-49f4-ab84-83302ff20010.jpeg',
+                'description' => 'test',
+                'copyrightHolder' => 'cc2',
+                'inLanguage' => 'nl',
+            ],
+            [
+                '@id' => 'https://io.uitdatabank.be/images/73585fe0-577b-488f-95f4-746940d7cce9',
+                '@type' => 'schema:ImageObject',
+                'contentUrl' => 'https://io.uitdatabank.be/images/73585fe0-577b-488f-95f4-746940d7cce9.jpeg',
+                'thumbnailUrl' => 'https://io.uitdatabank.be/images/73585fe0-577b-488f-95f4-746940d7cce9.jpeg',
+                'description' => 'test',
+                'copyrightHolder' => 'cc3',
+                'inLanguage' => 'nl',
+            ],
+        ],
+        'image' => 'https://io.uitdatabank.be/images/da02d848-eb11-4bfa-a566-d8bd3b856990.jpeg',
+    ];
+
+    private const POLYFILLED_JSON = [
+        'mediaObject' => [
+            [
+                '@id' => 'https://io.uitdatabank.be/images/da02d848-eb11-4bfa-a566-d8bd3b856990',
+                '@type' => 'schema:ImageObject',
+                'contentUrl' => 'https://images.uitdatabank.be/da02d848-eb11-4bfa-a566-d8bd3b856990.jpeg',
+                'thumbnailUrl' => 'https://images.uitdatabank.be/da02d848-eb11-4bfa-a566-d8bd3b856990.jpeg',
+                'description' => 'test',
+                'copyrightHolder' => 'cc2',
+                'inLanguage' => 'nl',
+            ],
+            [
+                '@id' => 'https://io.uitdatabank.be/images/21defff6-59e4-49f4-ab84-83302ff20010',
+                '@type' => 'schema:ImageObject',
+                'contentUrl' => 'https://images.uitdatabank.be/21defff6-59e4-49f4-ab84-83302ff20010.jpeg',
+                'thumbnailUrl' => 'https://images.uitdatabank.be/21defff6-59e4-49f4-ab84-83302ff20010.jpeg',
+                'description' => 'test',
+                'copyrightHolder' => 'cc2',
+                'inLanguage' => 'nl',
+            ],
+            [
+                '@id' => 'https://io.uitdatabank.be/images/73585fe0-577b-488f-95f4-746940d7cce9',
+                '@type' => 'schema:ImageObject',
+                'contentUrl' => 'https://images.uitdatabank.be/73585fe0-577b-488f-95f4-746940d7cce9.jpeg',
+                'thumbnailUrl' => 'https://images.uitdatabank.be/73585fe0-577b-488f-95f4-746940d7cce9.jpeg',
+                'description' => 'test',
+                'copyrightHolder' => 'cc3',
+                'inLanguage' => 'nl',
+            ],
+        ],
+        'image' => 'https://images.uitdatabank.be/da02d848-eb11-4bfa-a566-d8bd3b856990.jpeg',
+    ];
+
+    /**
+     * @test
+     */
+    public function it_should_replace_legacy_urls_when_enabled()
+    {
+        $id = '5624b810-c340-40a4-8f38-0393eca59bfe';
+        $mapping = [
```

Comment:
I would make this replace both UDB2 and UDB3 media urls, now it does not test it handles multiple replacements.

Reply:
Great suggestion. It made me find a bug in the mapping